### PR TITLE
Stop live555 logger from claiming it's backing up when it actually isn't

### PR DESCRIPTION
**Pull request by yousefg-codes**
Added a condition for backing up the logs (don't want to hard code the backup condition, will do that in the firmware repo), to prevent backup from occurring when it shouldn't.

### CODE REVIEW

File: .github/workflows/compile-check.yml [hunk 1]
```diff
@@ -21,4 +21,4 @@ jobs:
       run: ./genMakefiles linux
 
     - name: Make All live555 targets
-      run: make CPPFLAGS='-DLOG_FILE_TMP_DIR="\"/test/path\""'
+      run: make CPPFLAGS='-DDEBUG -DDEBUG_SEND'
```
Comment (asheffie1) [inline .github/workflows/compile-check.yml:24]: Looks like we're providing two new defines (`DEBUG` and `DEBUG_SEND`) we weren't previously providing - What do these do and why weren't we using them earlier?

File: UsageEnvironment/Log.cpp [hunk 2]
```diff
@@ -70,16 +70,36 @@ string LogLevelAsString(LogLevel logLevel) {
...
-void _Log::OutputToFile(const string filePath) {
-  currLogFilePath = filePath;
+_Log::_Log(const string logDirectory) : logLevel(LogLevelDebug) {
+  SetTmpLogDir(logDirectory);
 }
 
-void _Log::OutputToStdout(bool toStdout) {
-  logToStdout = toStdout;
+_Log::_Log() : logLevel(LogLevelDebug) {
+  SetTmpLogDir("");
```
Comment (asheffie1) [inline UsageEnvironment/Log.cpp]: Why do we have two constructors now? (i.e. Can't use default parameter where logDirectory is empty string?)

File: UsageEnvironment/Log.cpp [hunk 3]
```diff
@@ -70,16 +70,36 @@ string LogLevelAsString(LogLevel logLevel) {
...
+    FILE* logFileHandle = fopen(currLogFilePath.c_str(), "a");
+    if (logFileHandle == NULL) {
+      fprintf(stderr, "Couldn't acquire log number %u.%s\n", logFileNum, strerror(errno));
+    }
+    return logFileHandle;
+  }
+  fflush(stdout);
+}
+
+void _Log::SetTmpLogDir(std::string logDirectory) {
```
Comment (asheffie1) [inline UsageEnvironment/Log.cpp:77]: Nit: `SetTmpLogDir` function should be better positioned in the file. Where it is now causes an unnecessary diff of the `AcquireFile` function, making it look like that function was deleted in the original code and added somewhere else in the new code. This makes it harder to review changes.

File: UsageEnvironment/Log.cpp [hunk 4]
```diff
@@ -141,8 +146,14 @@ void _Log::NextLogFile() {
...
   lock_guard<mutex> backupLockGuard(logMutex);
+
+  if (!logFileDir.empty()) {
+    printf("Cannot backup live555 logs as it is not enabled.\n");
+    return;
+  }
+
   if (access(backupFilePath.c_str(), F_OK) == 0 && remove(backupFilePath.c_str()) < 0) {
-    fprintf(stderr, "Couldn't remove old backup file in %s%s\n", backupFilePath, strerror(errno));
+    fprintf(stderr, "Couldn't remove old backup file in %s%s\n", backupFilePath.c_str(), strerror(errno));
```
Comment (asheffie1) [inline UsageEnvironment/Log.cpp]: Wouldn't `%s%s` result in the error being appended to the file path as one long continuous line of text?

File: UsageEnvironment/include/Log.hh [hunk 5]
```diff
@@ -42,13 +42,13 @@ private:
...
-  _Log(const std::string logDirectory = "");
+  _Log(const std::string logDirectory);
+  _Log();
   ~_Log() {}
 
   void BackupLog(const std::string backupFileName);
-  void OutputToFile(const std::string filePath);
-  void OutputToStdout(bool toStdout);
   void SetLevel(LogLevel level);
+  void SetTmpLogDir(std::string logDirectory);
```
Comment (asheffie1) [inline UsageEnvironment/include/Log.hh:50]: Is it intentional to allow the client to change the log directory during runtime? If not, this should be a private function so client can't call it.

**Review by asheffie1 — CHANGES_REQUESTED** [review]

File: .github/workflows/compile-check.yml (hunk 1 above)
Comment (yousefg-codes) [reply .github/workflows/compile-check.yml:24]: This is just a simple github workflow I made to make sure that the branch always compiles without errors. These flags enable debug prints, but in this case I'm enabling them just to make sure any code that falls under them is compiled, these are already being used in the camera-firmware live555 bitbake though.

File: UsageEnvironment/include/Log.hh (hunk 5 above)
Comment (yousefg-codes) [reply UsageEnvironment/include/Log.hh:50]: Yes it's intentional. Changing it during runtime is what allows us to enable backing up and logging to a tmp directory. I left the compile-time flag in there just in case anyone wants to use it and doesn't want to use the SetTmpLogDir function.

**Review by adeel-verkada — APPROVED** [review]

File: .github/workflows/compile-check.yml (hunk 1 above)
Comment (asheffie1) [reply .github/workflows/compile-check.yml:24]: Ah makes sense, thanks for explaining.

File: UsageEnvironment/include/Log.hh (hunk 5 above)
Comment (asheffie1) [reply UsageEnvironment/include/Log.hh:50]: OK that fine, but in that case we need to protect `SetTmpLogDir()` function body with our mutex to be safe.

(I know in theory `SetTmpLogDir()` should be called before any debug prints happen, but we want to protect against future changes where things could get moved around and prints might happen concurrently with the client calling `SetTmpLogDir()`.)

**Review by asheffie1 — CHANGES_REQUESTED** [review]
